website/pages/docs/acl/auth-methods/oidc.mdx

```diff
@@ -0,0 +1,214 @@
+        },
+        "OIDCClientID": "V1RPi2MYptMV1RPi2MYptMV1RPi2MYpt",
+        "OIDCClientSecret": "...(omitted)...",
+        "OIDCDiscoveryURL": "https://my-corp-app-name.auth0.com/"
+    }
+}
+```
+
+## JWT Verification
+
+JWT signatures will be verified against public keys from the issuer via OIDC
+discovery. Keys will be fetched from the OIDC Discovery URL during
+authentication and OIDC validation criteria (e.g. `iss`, `aud`, etc.) will be
+applied.
+
+## OIDC Authentication
+
+Consul includes two built-in OIDC login flows: the Consul UI, and the CLI using
+[`consul login`](/docs/commands/login).
+
+### Redirect URIs
+
+An important part of OIDC auth method configuration is properly setting
+redirect URIs. This must be done both in Consul and with the OIDC provider, and
+these configurations must align. The redirect URIs are specified for an auth
+method with the [`AllowedRedirectURIs`](#allowedredirecturis) parameter. There
+are different redirect URIs to configure the Consul UI and CLI flows, so one or
+both will need to be set up depending on the installation.
+
+#### Consul UI
+
+Logging in via the Consul UI requires a redirect URI of the form:
+`http://localhost:8500/ui/oidc/callback` or
+`https://{host:port}/ui/oidc/callback`
+
+The "host:port" must be correct for the Consul server.
+
+#### CLI
+
+If you plan to support authentication via `consul login -type=oidc
+-method=<name>`, a localhost redirect URI must be set (usually this is
+`http://localhost:8550/oidc/callback`). Logins via the CLI may specify a
+different host and/or listening port if needed, and a URI with this host/port
+must match one of the configured redirected URIs. These same "localhost" URIs
+must be added to the provider as well.
+
+### OIDC Login
+
+#### Consul UI
+
+**TODO(sso): update this after john is done**
+
+1. Select the "OIDC" login method.
+2. Enter a role name if necessary.
+3. Press "Sign In" and complete the authentication with the configured provider.
+
+#### CLI
+
+```
+$ consul login -method=oidc -type=oidc -token-sink-file=consul.token
+
+Complete the login via your OIDC provider. Launching browser to:
+
+    https://myco.auth0.com/authorize?redirect_uri=http%3A%2F%2Flocalhost%3A8550%2Foidc%2Fcallback&client_id=r3qXc2bix9eF...
+```
+
+The browser will open to the generated URL to complete the provider's login.
+The URL may be entered manually if the browser cannot be automatically opened.
+
+The callback listener may be customized with the following optional parameters.
+These are typically not required to be set:
+
+The callback listener defaults to listen on `localhost:8550`. If you want to
+customize that use the optional flag
+[`-oidc-callback-listen-addr=<host:port>`](/docs/commands/login#oidc-callback-listen-addr).
+
+## OIDC Configuration Troubleshooting
+
+The amount of configuration required for OIDC is relatively small, but it can
+be tricky to debug why things aren't working. Some tips for setting up OIDC:
```

Comment:
Love this section! ❤️

Reply:
SOOOOOO much of this is directly lifted from vault's docs since they already presumably have had to field customer debugging sessions already.